webdriver/tests/classic/perform_actions/key.py

```diff
@@ -24,10 +24,17 @@ def test_no_browsing_context(session, closed_frame, key_chain):
 def test_key_down_closes_browsing_context(
     session, configuration, http_new_tab, inline, key_chain
 ):
-    session.url = inline("""
+    url = inline("""
         <input onkeydown="window.close()">close</input>
         <script>document.querySelector("input").focus();</script>
         """)
+
+    # Open a new window.
+    new_window = session.execute_script(f"return window.open('{url}')")
+
+    # Switch to the new window.
+    session.window_handle = new_window.id
```

Comment:
This is still failing Firefox tests because none of the tests are waiting for the navigation to be completed. We need to wait for the `input` element to have the focus before we can start performing the action.